spec.bs

```diff
@@ -6,57 +6,940 @@ Status: CG-DRAFT
+    1. If |shed|[origin] does not exist, then set |shed|[origin] to the result of running [=create a shared storage shelf=] with [=storage type|type=] "`shared`".
+    1. Return |shed|[|origin|].
+  </div>
+
+  <div algorithm>
+    To <dfn>create a shared storage shelf</dfn>, run these steps:
+
+    1. Let |shelf| be a new [=storage shelf=].
+    1. Set |shelf|'s [=bucket map=]["`default`"] to the result of running [=create a shared storage bucket=].
+    1. Return |shelf|.
+  </div>
+
+  A <dfn>shared storage bucket</dfn> is a [=storage bucket=] in one of a [=shared storage shed=]'s [=storage shelf|shelves=].
+
+  <div algorithm>
+    To <dfn>create a shared storage bucket</dfn>, run these steps:
+
+    1. Let |endpoint| be the [=storage endpoint=] with [=storage identifier=] "`sharedStorage`".
+    1. Let |bucket| be a new [=shared storage bucket=].
+    1. Set |bucket|'s [=bottle map=]["`sharedStorage`"] to a new [=storage bottle=] whose [=storage bottle/quota=] is |endpoint|'s [=storage endpoint/quota=].
+    1. Return |bucket|.
+  </div>
+
+  Note: Currently, a [=shared storage bucket=]'s [=bottle map=] has [=map/size=] `1`, since there is only one [=storage endpoint=] [=storage endpoint/registered=] with [=storage type|type=] "`shared`".
+
+  <div algorithm>
+    To <dfn>obtain a shared storage bottle map</dfn>, given an [=environment settings object=] |environment|, run these steps:
+
+    1. Let |shed| be the [=user agent=]'s [=shared storage shed=].
+    1. Let |shelf| be the result of running [=obtain a shared storage shelf=] with |shed| and |environment|.
+    1. If |shelf| is failure, then return failure.
+    1. Let |bucket| be |shelf|'s [=bucket map=]["`default`"].
+    1. Let |bottle| be |bucket|'s [=bottle map=]["`sharedStorage`"].
+    1. Let |proxyMap| be a new [=storage proxy map=] whose [=backing map=] is |bottle|'s [=map=].
+    1. [=set/Append=] |proxyMap| to |bottle|'s [=proxy map reference set=].
+    1. Return |proxyMap|.
+  </div>
+
+  ## The [=Shared Storage Database=] ## {#database}
+
+  A [=/browsing context=] has an associated <dfn>shared storage database</dfn>, which provides methods to [=shared storage database/store an entry in the database|store=], [=shared storage database/retrieve an entry from the database|retrieve=], [=shared storage database/delete an entry from the database|delete=], [=shared storage database/clear all entries in the database|clear=], and [=shared storage database/purge expired entries from the database|purge expired=] data, and additional methods as below. The data in the [=shared storage database|database=] take the form of [=shared storage database/entry|entries=].
+
+  Each [=shared storage database=] has a <dfn for="shared storage database">shared storage database queue</dfn>, which is a [=parallel queue=] on which the each of the [=shared storage database=]'s methods run when calls to them are initiated from that [=/browsing context=].
```

Comment:
I think you need to say the queue is the resulting of starting the parallel queue:

https://html.spec.whatwg.org/multipage/infrastructure.html#starting-a-new-parallel-queue